Pokedex detail rework

madninjaskillz opened the pull request:
Closes issues
-------------
#814

Changes
-------
- Moves Pokedex details to its own page
- Pokedex details displayed on FlipView to support swipes

Change details
--------------

Major refactor of the pokedex code - introduces some new more generic models.

![wp_ss_20160902_0002](https://cloud.githubusercontent.com/assets/2499871/18217381/5c9327b8-7154-11e6-95cc-7edbdcaf93c7.png)

![wp_ss_20160902_0004](https://cloud.githubusercontent.com/assets/2499871/18217341/29867b4a-7154-11e6-8ee3-dd05fabd8d0e.png)

Rover656 commented:
Sounds good, what are the conflicts?

madninjaskillz commented:
honestly dont know - this is my first proper GitHub pull request. I can code, don't know how to GitHub yet!

Seelengrab commented:
Ok, there's quite a bit wrong (or rather, not like we have agreed on) with this PR.

1. Move the model you created to entities. That's where all the models are located right now.
2. Fix your formatting. You have mixed tabs & spaces. No dangling `/>` either please. 
3. Code should be indented according to [Allman style](https://en.wikipedia.org/wiki/Indent_style#Allman_style) (4 spaces per indentation level). You can set this to be done automatically in VS settings per language. Make sure the first property for a XAML element is inline with the opening brackets and align further properties with the first one, like this: 

``` 
<TextBlock FontSize="20"
                   FontWeight="Bold" />
```

4. Remove commented out sections of code you do not use. Git keeps track of old code, no need to keep it around if you don't use it.
5. Remove unnecessary whitespace or unnecessary empty lines, such as line 29 in PokedexPageViewModel.cs or line 65 through 68 in SettingsPageViewModel.cs

madninjaskillz commented:
I'll do my best on fixing the formatting in XAML - but most of it is just lifted from what was already there (and tbh I'm more likely to rewrite it than to fix the formatting because its a nightmare)

I'm a little confused by some of the comments - I've never even opened SettingsPageViewModel (nor is in in this pull) for instance - I don't see how my formatting could be bad there. I'm happy to spend some time fixing formatting in other stuff unrelated to this pull but I don't see why it should effect this pull.

as for the commented out code; I'm ashamed, that's a pet peeve of mine ;)

madninjaskillz commented:
hopefully that should be a little better.

Seelengrab commented:
Hm.. I'm pretty sure it was in the changed files of this PR :o Anyway, I'll take a look at it.

Seelengrab commented:
Ok, just a few points:

- Please don't leave an empty `catch` (DataTemplateSelectors.cs / Line 31). This might be good when you're the only developer and you know what exception can occur and why it can be ignored - however, this is a community effort, so logging the exception or commenting why you do nothing when an exception occurs is the least you can do. That way, a different developer is able to trace it in case something happens when debugging.
- Curly brackets are only necessary in multiline statements - if-statements containing only one line (such as `break;`) are fine without them. This is just a note for the feature though, no need to change that back again. I'll take a note for the refactor about it.
- Take another look at PokedexDetailPage.xaml; The indentation of some elements is still wonky and having all properties on the same line as the declaration of an element is really hard to parse visually. Please move each property to its own line, with the first property being inline with the opening tag of the element, like this:

    ``` 
    <TextBlock FontSize="20"
               FontWeight="Bold" />
    ```

- Custom resources like styles and templates are all contained in one shared resource dictionary. It can be found at Styles/Custom.xaml, please move yours there.

Other than that, looks pretty clean to me.

madninjaskillz commented:
fixed the first point, second point is an issue I fight all the time ;) I KNOW you CAN have single line if's without your curlys, i just dont like them - i prefer to have the scope (both variable scope and execution scope) very clearly defined.

fixed up the xaml some more.

I've left the templates where they are as I just couldn't get them to work after moving.

Seelengrab commented:
LGTM

ST-Apps commented:
Thanks for your help and sorry for the messy code that you had to work on, but this happens when a lot of people contribute without any kind of guidelines. Hopefully 2.0 version will be a lot better :)
